### PR TITLE
feat: add struct support to expr printer and `DocAst`

### PR DESCRIPTION
**Pull request by dghosef**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala [hunk 1]
```diff
@@ -127,6 +127,12 @@ object DocAst {
 
     case class Native(clazz: Class[_]) extends Atom
 
+    case class StructNew(sym: Symbol.StructSym, exps: List[(Name.Label, Expr)], exp: Expr) extends Composite
+
+    case class StructGet(exp: Expr, field: Name.Label, tpe: Type) extends Composite
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:132]: DocAst just have to look right, so get and put probably doesnt have to be case classes.

File: main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala [hunk 2]
```diff
@@ -127,6 +127,8 @@ object DocAst {
 
     case class Native(clazz: Class[_]) extends Atom
 
+    case class StructNew(sym: Symbol.StructSym, exps: List[(Name.Label, Expr)], exp: Expr) extends Composite
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:130]: Sorry for the slow review

This should be expressible with `DoubleKeyword("new", exps.foldRight(RecordEmpty: Expr){ case (acc, cur) => RecordExtend(cur._1, cur._2, acc)}, "@", d2)`

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:130]: or something close to that

Comment (dghosef) [reply main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:130]: I would need `new StructName {...} @ r`. Can I just concatenate the struct name after the "new" in `DoubleKeyword`?

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:130]: oh yeah - yes that works

File: main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala [hunk 3]
```diff
@@ -204,6 +204,15 @@ object DocAst {
     def ArrayStore(d1: Expr, index: Expr, d2: Expr): Expr =
       Assign(SquareApp(d1, List(index)), d2)
 
+    def StructNew(sym: Symbol.StructSym, exps: List[(Name.Label, Expr)], d2: Expr) =
+      DoubleKeyword("new " + sym.toString, exps.foldRight(RecordEmpty: Expr){ case (cur, acc) => RecordExtend(cur._1, cur._2, acc)}, "@", Left(d2))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/dbg/DocAst.scala:208]: Maybe one or more local variables would be reasonable here.

Also the return type of the function is missing.

**Comment by magnus-madsen** [comment]
@JonathanStarup Is this OK?

**Comment by JonathanStarup** [comment]
looks good, though its probably good to check the output manually, no tests touch this code

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**